### PR TITLE
Add action to test pipeline runs successfully

### PR DESCRIPTION
**Pull request by adamjtaylor**
This PR adds an action that runs the nf-artist pipeline on a samplesheet of two test datasets. Each requires conversion and makes a Minerva story and a Miniature thumbnail. One is a H&E image and the other is a CyCIF image. This action would be required to be passed for the PR to be merged.

### CODE REVIEW

File: .github/workflows/nextflow.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    branches: 'main'
+  workflow_dispatch:
+
+jobs:
+  nf-artist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: nf-core/setup-nextflow@v1
+      - run: nextflow run ${GITHUB_WORKSPACE} --input data/test_samplesheet.csv
```
Comment (thomasyu888) [inline .github/workflows/nextflow.yml:14]: This is great!  We may want to think about having a "test" nextflow config that would just have this - that said, its unnecessary here.

File: .github/workflows/nextflow.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  pull_request:
+    branches: 'main'
+  workflow_dispatch:
+
+jobs:
+  nf-artist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (thomasyu888) [inline .github/workflows/nextflow.yml:12]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/nextflow.yml (hunk 1 above)
Comment (adamjtaylor) [reply .github/workflows/nextflow.yml]: Thats a great suggestion. I have added

**Review by thomasyu888 — APPROVED** [review]
Excellent!

**Pull request merged**